script/gen/cxx14/all

```diff
@@ -0,0 +1,419 @@
+                sections[section].append(spec)
+    return sections
+
+def write_header_prologue(filep):
+    """ Writes the prologue of the header file """
+    filep.write("// Auto-generated by %s; don't edit\n" % sys.argv[0])
+    filep.write("#ifndef MEASUREMENT_KIT_CXX14_ALL_HPP\n")
+    filep.write("#define MEASUREMENT_KIT_CXX14_ALL_HPP\n")
+    filep.write("\n")
+    filep.write("#include <assert.h>\n")
+    filep.write("#include <stdint.h>\n")
+    filep.write("\n")
+    filep.write("#include <functional>\n")
+    filep.write("#include <memory>\n")
+    filep.write("#include <string>\n")
+    filep.write("#include <thread>\n")
+    filep.write("#include <utility>\n")
+    filep.write("#include <vector>\n")
+    filep.write("\n")
+    filep.write("#include <measurement_kit/common/nlohmann/json.hpp>\n")
+    filep.write("#include <measurement_kit/ffi.h>\n")
+    filep.write("\n")
+    filep.write("namespace mk {\n")
+    filep.write("namespace cxx14 {\n")
+    filep.write("\n")
+
+def write_ffi_helpers(filep):
+    """ Writes helpers to handle FFI types """
+    filep.write("/// Deletes a mk_task_t pointer\n")
+    filep.write("class TaskDeleter {\n")
+    filep.write("  public:\n")
+    filep.write("    void operator()(mk_task_t *task) noexcept { mk_task_destroy(task); }\n")
+    filep.write("};\n")
+    filep.write("\n")
+    filep.write("/// Unique pointer to mk_task_t\n")
+    filep.write("using TaskUptr = std::unique_ptr<mk_task_t, TaskDeleter>;\n")
+    filep.write("\n")
+    filep.write("/// Deletes a mk_event_t pointer\n")
+    filep.write("class EventDeleter {\n")
+    filep.write("  public:\n")
+    filep.write("    void operator()(mk_event_t *event) noexcept { mk_event_destroy(event); }\n")
+    filep.write("};\n")
+    filep.write("\n")
+    filep.write("/// Unique pointer to mk_event_t\n")
+    filep.write("using EventUptr = std::unique_ptr<mk_event_t, EventDeleter>;\n")
+    filep.write("\n")
+
+def write_events_classes(filep, events):
+    """ Writes event classes on the header file """
+    for key, value in sorted(events.items()):
+        filep.write("/// C++ representation of '%s' event\n" % key)
+        filep.write("class %s {\n" % event_name_to_camel_case(key))
+        filep.write("  public:\n")
+        for name, pattern in value.items():
+            filep.write("    ")
+            if isinstance(pattern, STR):
+                filep.write("std::string %s;\n" % name)
+            elif isinstance(pattern, int):
+                filep.write("int64_t %s = 0;\n" % name)
+            elif isinstance(pattern, float):
+                filep.write("double %s = 0.0;\n" % name)
+            else:
+                raise RuntimeError("unexpected type for: %s" % pattern)
+        filep.write("};\n\n")
+
+def write_events_callbacks(filep, events):
+    """ Writes events callbacks in the header file """
+    for key, value in sorted(events.items()):
+        filep.write("/// Callback taking `const %s &` as unique argument\n" %
```

Comment:
All these doxygen comments lead me to think that:

1. probably the main README should point to doxygen rather than replicating synopsis

2. more explicit documentation can be moved from engine.hpp to ffi.h

3. there should be automatic codedocs documentation also for sub APIs

4. we should not install README.md:s as part of `make install`

5. we probably need other README.md:s for other sub APIs

6. we should make sure the list of other APIs in the main README.md for include is up-to-date